app/src/main/java/net/opendasharchive/openarchive/db/Space.kt

```diff
@@ -1,10 +1,29 @@
 package net.opendasharchive.openarchive.db
 
+import android.content.Intent
 import android.text.TextUtils
+import androidx.appcompat.app.AppCompatActivity
 import com.orm.SugarRecord
+import net.opendasharchive.openarchive.features.onboarding.SpaceSetupActivity
 import net.opendasharchive.openarchive.util.Constants.EMPTY_STRING
 import net.opendasharchive.openarchive.util.Prefs
 
+class SpaceChecker {
```

Comment:
I'm not too happy with the wording here, both the class and the function. They're kind of irritating and don't really tell what they do.

If you have names, which don't convey meaning properly, it's typically a sign, that the architecture isn't at a quality position, yet where it should be.

Reply:
There was an existing class called ``Space`` within the project. And since the idea behind this was to check if user has logged into a Space, I named it as such (``SpaceChecker``). 

For the function name, I feel ``navigateToHome`` is rather appropriate and self explanatory. I could potentially rename it to ``navigateToLogin`` or ``navigateToLoginIfNoSpaceExists`` if you feel otherwise.

Open to suggestions to both.

Reply:
The problem is in the calling code.

One will read `SpaceChecker.navigateToHome()`.

That's just plain confusing. I appreciate the refactoring, but I think, this is actually too much.

There should rather be something like

```
if no space {
  go login
}
else {
  continue with what user wanted to do.
}
```

However, see my other comment. I think this is the wrong approach. I suggest, that @johnhess changes the scope of this.